File: Sección02_Regresión/PreProcesado de Datos/missing_data.R
#Plantilla para Pre Procesado de datos - Datos Faltantes

#Importar el dataset
dataset = read.csv('Data.csv')

#Tratamiendo de los valores NAs
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)
dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)

File: Sección02_Regresión/PreProcesado de Datos/missing_data.py
# -*- coding: utf-8 -*-
"""
Created on Fri Mar 17 23:50:12 2023

@author: Ckonny
"""

# Importar las librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el dataset
dataset = pd.read_csv('Data.csv')
X =     dataset.iloc[:,:-1].values
y =     dataset.iloc[:,3].values

# Tratamiento de los NAs
from sklearn.impute import SimpleImputer
imputer = SimpleImputer(missing_values = np.nan, strategy = 'mean')
imputer = imputer.fit(X[:, 1:3]) 

# Sobreescribimos nuestro imputer con los nuevos datos
X[:,1:3] = imputer.transform(X[:,1:3])